src/diracx/db/sandbox_metadata/db.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import datetime
+
+import sqlalchemy
+
+from ..utils import BaseDB
+from .schema import Base as SandboxMetadataDBBase
+from .schema import sb_Owners, sb_SandBoxes
+
+
+class SandboxMetadataDB(BaseDB):
+    # This needs to be here for the BaseDB to create the engine
+    metadata = SandboxMetadataDBBase.metadata
+
+    async def _get_put_owner(self, owner: str, owner_group: str) -> int:
+        """adds a new owner/ownerGroup pairs, while returning their ID if already existing
+
+        Args:
+            owner (str): _description_
+            owner_group (str): _description_
+        """
+        stmt = sqlalchemy.select(sb_Owners.OwnerID).where(
+            sb_Owners.Owner == owner, sb_Owners.OwnerGroup == owner_group
+        )
+        result = await self.conn.execute(stmt)
+        res = result.fetchall()
+        if res:
+            return res[0][0]
+
+        stmt = sqlalchemy.insert(sb_Owners).values(Owner=owner, OwnerGroup=owner_group)
+        result = await self.conn.execute(stmt)
+        return result.lastrowid
+
+    async def insert(
+        self, owner: str, owner_group: str, sb_SE: str, se_PFN: str, size: int = 0
+    ) -> tuple[int, bool]:
+        """_insert a new sandbox
+        this is "equivalent" of DIRAC registerAndGetSandbox
+
+        Args:
+            owner (str): _description_
+            owner_group (str): _description_
+            sb_SE (str): _description_
+            sb_PFN (str): _description_
+            size (int, optional): _description_. Defaults to 0.
+        """
+        owner_id = await self._get_put_owner(owner, owner_group)
+        stmt = sqlalchemy.insert(sb_SandBoxes).values(
```

Comment:
I think you can get rid of the `# type: ignore` comments if you use:
```suggestion
        stmt: sqlalchemy.Executable = sqlalchemy.insert(sb_SandBoxes).values(
```

An aside (which hopefully isn't needed at all in this case), it's probably better to stick to `# type: ignore[no-redef]` rather then the catch all `# type: ignore` comment so it's self documenting.